src/transformers/models/sam_hq/processing_samhq.py

```diff
@@ -0,0 +1,327 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for SAMHQ.
+"""
+
+from copy import deepcopy
+from typing import List, Optional
+
+import numpy as np
+
+from ...image_utils import ImageInput
+from ...processing_utils import ImagesKwargs, ProcessingKwargs, ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding
+from ...utils import is_torch_available
+
+
+if is_torch_available():
+    import torch
+
+
+class SamHQImagesKwargs(ImagesKwargs):
+    segmentation_maps: Optional[ImageInput]
+    input_points: Optional[List[List[float]]]
+    input_labels: Optional[List[List[int]]]
+    input_boxes: Optional[List[List[List[float]]]]
+    point_pad_value: Optional[int]
+
+
+class SamHQProcessorKwargs(ProcessingKwargs, total=False):
+    images_kwargs: SamHQImagesKwargs
+    _defaults = {
+        "images_kwargs": {
+            "point_pad_value": None,
+        }
+    }
+
+
+class SamHQProcessor(ProcessorMixin):
+    r"""
+    Constructs a SAM HQ processor which wraps a SAM  image processor and an 2D points & Bounding boxes processor into a
+    single processor.
+
+    [`SamHQProcessor`] offers all the functionalities of [`SamImageProcessor`]. See the docstring of
+    [`~SamImageProcessor.__call__`] for more information.
+
+    Args:
+        image_processor (`SamImageProcessor`):
+            An instance of [`SamImageProcessor`]. The image processor is a required input.
+    """
+
+    attributes = ["image_processor"]
+    image_processor_class = "SamImageProcessor"
+
+    optional_call_args = [
+        "segmentation_maps",
+        "input_points",
+        "input_labels",
+        "input_boxes",
+    ]
+
+    def __init__(self, image_processor):
+        super().__init__(image_processor)
+        # Ensure image_processor is properly initialized
+        if not hasattr(self, "image_processor"):
+            raise ValueError("image_processor was not properly initialized")
+        if not hasattr(self.image_processor, "size"):
+            raise ValueError("image_processor.size is not set")
+        self.target_size = self.image_processor.size["longest_edge"]
+
+    def __call__(
+        self,
+        images: Optional[ImageInput] = None,
+        # The following is to capture `segmentation_maps`, `input_points`, `input_labels` and `input_boxes`
+        # arguments that may be passed as a positional argument.
+        # See transformers.processing_utils.ProcessorMixin.prepare_and_validate_optional_call_args for more details,
+        # or this conversation for more context:
+        # https://github.com/huggingface/transformers/pull/32544#discussion_r1720208116
+        # This behavior is only needed for backward compatibility and will be removed in future versions.
+        *args,  # to be deprecated
+        **kwargs,
```

Comment:
This was removed in a previous commit I think - kwargs have to be typed! It's the purpose of the TypedDict defined above. You're missing the import of `Unpack` from `processing_utils` as well, and then you have to add this.  
 
```suggestion
        **kwargs: Unpack[SamHQProcessorKwargs],,
```

Reply:
done